### PR TITLE
Ncnews 10

### PR DESCRIPTION
**Pull request by NookaCoder**
Adds endpoint to /api/articles/:article_id/comments
Adds POST 200, which adds comment and returns new comment
Adds POST 400 errors for blank objects and invalid ids
Adds POST 404 errors for non-existent ids and usernames

### CODE REVIEW

File: __tests__/app.test.js [hunk 1]
```diff
@@ -199,6 +199,77 @@ describe('/api/articles/:article_id/comments', () => {
...
+            'body': 'It were reet'
+        };
+        return request(app)
+            .post('/api/articles/3/comments')
+            .send(newComment)
+            .expect(404)
+            .then((response) => {
+                expect(response.body.msg).toBe('ID does not exist')
+            })
+    });
```
Comment (Poonam-raj) [inline __tests__/app.test.js:286]: what if username or body were missing - I know you have the test on line 237 for an empty object, would that work for just one missing property too?

File: controllers/topics-controllers.js [hunk 2]
```diff
@@ -54,6 +55,25 @@ exports.getCommentsByArticleId = (req, res, next) => {
...
+        insertComment(newComment).then((comment) => {
+            res.status(201).send({ comment });
+        }).catch((err) => {
+            next(err);
+        });
+    } else {
+        res.status(400).send({
+            'msg': 'Comment undefined'
+        });
+    }
```
Comment (Poonam-raj) [inline controllers/topics-controllers.js]: I would consider letting your error handling middleware cover the error response on line 71 rather than directly sending it in your controller. to let your middleware handle it you'd have to promise.reject in your model and let the controller catch it to pass it elsewhere. This would mean you'd need to move your logic (that checks if the comment body is defined) out the controller and into the model

File: models/topics-models.js [hunk 3]
```diff
@@ -42,6 +42,15 @@ exports.selectCommentsByArticleId = (article_id) => {
         });
 };
 
+exports.insertComment = (newComment) => {
+    const { username, body, article_id } = newComment;
```
Comment (Poonam-raj) [inline models/topics-models.js:46]: I'd validate the request body here

**Review by Poonam-raj — CHANGES_REQUESTED** [review]
bit of amendment needed - I'd like to see another PR to double check you refactored your controller appropriately. I'd like any 'rejecting' to happen in the model ideally and let the controller just pass errors onto the error handling middleware

File: __tests__/app.test.js [hunk 4]
```diff
@@ -199,6 +199,91 @@ describe('/api/articles/:article_id/comments', () => {
                 expect(response.body.msg).toBe('Article does not exist')
             })
     });
+    test('POST:200 adds comment and responds with posted comment', () => {
+        const newComment = {
+            'username': 'jessjelly',
+            'body': 'It were reet'
```
Comment (dappernerddesigns) [inline __tests__/app.test.js:205]: Love this 😆

File: __tests__/app.test.js [hunk 5]
```diff
@@ -199,6 +199,91 @@ describe('/api/articles/:article_id/comments', () => {
...
+        return request(app)
+            .post('/api/articles/3/comments')
+            .send(blankBody)
+            .expect(400)
+            .then(({body}) => {
+                expect(body.msg).toBe("Comment undefined")
+            });
+    });
+    test('POST:404 sends error message when no username provided', () => {
+        const blankUser = {
```
Comment (dappernerddesigns) [inline __tests__/app.test.js:250]: If no username is provided, I would argue this should be a 400 like the others above

**Review by dappernerddesigns — APPROVED** [review]
Just one minor change on a test, swapping a 404 for a 400 :) Make the change, pass the test then merge and move on. Great work Carlie.
